pinot-core/src/test/java/com/linkedin/pinot/core/segment/store/SegmentDirectoryPathsTest.java

```diff
@@ -34,4 +38,50 @@ public void testSegmentPathFor() {
     Assert.assertEquals(v2Dir, f);
     Assert.assertEquals(v3Dir, new File("f", SegmentDirectoryPaths.V3_SUBDIRECTORY_NAME));
   }
+
+  @Test
+  public void testFindMetadataFile()
+      throws IOException {
+    File tempDirectory = null;
+    try {
+      // setup temp dir, v3 subdir and create metadata.properties in both
+      tempDirectory = new File(SegmentDirectoryPaths.class.toString());
+      FileUtils.forceMkdir(tempDirectory);
```

Comment:
added it but I think that is not required.
If jvm is killed then deleteOnExit will not have chance to run. If there is any exception then finally will execute. I had used deleteOnExit only in the past but that fails if the directory is not empty.